### PR TITLE
Add Bitdefender SQLite Map

### PR DESCRIPTION
**Pull request by anelshaer**
## Description

Adding a Map for Bitdefender SQLite DB. 

## Checklist:
Please replace every instance of `[ ]` with `[X]`

- [X] I have generated a unique GUID for my Map file(s)
- [X] I have tested and validated the new Map file(s) work with my test data and achieve the desired output
- [X] I have placed the Map file(s) within the .\SQLMap\Maps directory
- [X] I have set or updated the version of my Maps file(s)
- [X] I have made an attempt to document the artifacts within the Maps file(s)

Thank you for your submission and for contributing to the DFIR community!

### CODE REVIEW

**Comment by rathbuna** [comment]
Thank you for the contribution! Nice work! Quick question, are we sure the DB is called RansomewareRecover.db? Ransomware doesn't have an `e` in it between the `m` and `w`. 

Also, since SQLECmd is a multi OS tool, I added `Windows_` to the beginning of each filename.

**Comment by anelshaer** [comment]
Thanks a lot, my bad, i confirmed the file name and corrected the map name.

**Comment by rathbuna** [comment]
Okay thank you. Do you need to test that Map against your test data again to make sure it works? It surely wouldn't have worked with the wrong DB included in the Map.

File: SQLMap/Maps/Windows_Bitdefender_Antiphishing.smap [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                result AS Result,
+                datetime( expire / 1000, 'unixepoch', 'localtime' ) AS ExpireTime
+                FROM
+                aph_cache
+                ORDER BY
+                ExpireTime ASC;
+        BaseFileName: Antiphishing
+
+# Documentation
+# Use SQLECmd in conjunction with the Chrome KAPE Target: https://github.com/EricZimmerman/KapeFiles/blob/master/Targets/Antivirus/Bitdefender.tkape
```
Comment (rathbuna) [inline SQLMap/Maps/Windows_Bitdefender_Antiphishing.smap:25]: Change Chrome :)

File: SQLMap/Maps/Windows_Bitdefender_RansomwareRecover.smap [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                datetime( files.insert_time / 1000, 'unixepoch', 'localtime' ) AS InsertTime,
+                datetime( files.last_operation_time / 1000, 'unixepoch', 'localtime' ) AS LastOperationTime
+                FROM
+                files INNER JOIN packs ON files.packid = packs.uuid
+                ORDER BY
+                LastOperationTime ASC;
+        BaseFileName: RansomwareRecover
+
+# Documentation
+# Use SQLECmd in conjunction with the Chrome KAPE Target: https://github.com/EricZimmerman/KapeFiles/blob/master/Targets/Antivirus/Bitdefender.tkape
```
Comment (rathbuna) [inline SQLMap/Maps/Windows_Bitdefender_RansomwareRecover.smap:32]: @anelshaer here too

**Comment by anelshaer** [comment]
Thanks a lot for noting that, I corrected all that, I have tested the data as well. for ransomware, i don't have any data in the DB and here it shows it been processed

```
Processing 'C:\Users\User1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\RansomwareRecover.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        Map queries found: 1. Processing queries...
         'Bitdefender RansomewareRecover DB Files' did not return any results. CSV will not be saved.
```

**Comment by rathbuna** [comment]
> 
> 
> Thanks a lot for noting that, I corrected all that, I have tested the data as well. for ransomware, i don't have any data in the DB and here it shows it been processed
> 
> ```
> Processing 'C:\Users\User1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\RansomwareRecover.db'...
>         For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
>         For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
>         Map queries found: 1. Processing queries...
>          'Bitdefender RansomewareRecover DB Files' did not return any results. CSV will not be saved.
> ```

Thank you for providing that, very helpful. What this is saying is that it's looking to match the table(s) listed in the Map against the schema of the SQLite Database, so for instance, if your Map says look for a table called Hello and a table called World, your IdentifyValue would be 2 because you want the DB to match the 2 tables listed in the Map on the IdentifyQuery line so the right Map is being used against the right DB.

The output above says it's being told it has to look for 1 (which your Map has IdentifyValue as 2, which doesn't make sense?) but it found 0, meaning that there is a discrepancy somewhere but not in the SQLite syntax, because SQLECmd would've thrown an error and given you hints as to where the error is. Verify the content of the RansomwareRecover.db in a SQLite DB viewer and confirm which tables exist in the DB, and make sure your Map is calling foher those tables with the correct name(s) in the IdentityQuery line and number of table(s) in the IdentityValue line. 

Hopefully that makes sense. We'll get this figured out! Worst case scenario, can you provide me your DB ad I can take a look?

**Comment by anelshaer** [comment]
I have already made sure of all that, and there are no results because my DB instance RansomwareRecover is empty.
I have tested that IdentityQuery in SQL viewer and it indeed shows 2, I think it's something in the tool output itself but it's working as expected because the other 2 files I have data and they actually generating CSV files. 


```
.\Modules\bin\SQLECmd\SQLECmd.exe -d "C:\Users\user1\Desktop\testkape\C" --hunt --csv "C:\Users\user1\Desktop\testkape\" |  findstr.exe /i 'Bitdefender'
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\Antiphishing.db'...
        Dumping 'Bitdefender Antiphishing DB' to 'C:\Users\user1\Desktop\testkape\20210523200630653752_Bitdefender_Antiphishing_c3a9343a-cd65-484e-a8d2-b4e69b9924a1.csv'
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\ecal.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
        No maps found for 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\ecal.db'. Adding to unmatched database list
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\es.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        Dumping 'Bitdefender es DB' to 'C:\Users\user1\Desktop\testkape\20210523200630950297_Bitdefender_es_b5341baa-9fa9-4761-9a86-4fca69aca6e8.csv'
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\RansomwareRecover.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
         'Bitdefender RansomewareRecover DB Files' did not return any results. CSV will not be saved.
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\cryptoStorage\crypt.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
        No maps found for 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\cryptoStorage\crypt.db'. Adding to unmatched database list
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\events\events.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
        No maps found for 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\events\events.db'. Adding to unmatched database list
Skipping 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\secure_efs\EFSBackup.db' as a file with SHA-1 'B3215DD97B4DFE73577E2CA86A307073D97EBD0D' has already been processed
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\cloudprocesses.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
        No maps found for 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\cloudprocesses.db'. Adding to unmatched database list
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\conference_apps.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
        No maps found for 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\conference_apps.db'. Adding to unmatched database list
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\optimizer\UserProcesses.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
        No maps found for 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\optimizer\UserProcesses.db'. Adding to unmatched database list
        File name: 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\ecal.db', Tables: Certificate1,ClientHello2,DeviceDetection,DnsQueries,EventsConfig,GatheredQids,LastSentQid,SSLScore1,ServerHello2,sqlite_sequence
        File name: 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\cryptoStorage\crypt.db', Tables: dat,ref
        File name: 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\events\events.db', Tables: events
        File name: 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\cloudprocesses.db', Tables: Processes,android_metadata,sqlite_sequence
        File name: 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\conference_apps.db', Tables: conference_apps
        File name: 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\settings\optimizer\UserProcesses.db', Tables: Processes,sqlite_sequence
```

**Comment by rathbuna** [comment]
Hmm I emptied out a database and ran a Map against it and got the following output:

```
Checking if 'V:\SQL DBs\Mine\History' is a SQLite file
'V:\SQL DBs\Mine\History' is a SQLite file!
Adding 'V:\SQL DBs\Mine\History' SHA-1 '72888E10880B43947FB961971045E419BB3E56CA' to seen hashes collection
Processing 'V:\SQL DBs\Mine\History'...
Processing map 'Google Chrome Downloads' with Id '0b575bcf-ade3-44e5-8162-eb52a96c2b06'
        Verifying database via 'SELECT count(*) FROM sqlite_master WHERE type='table' AND (name='urls' OR name='visits' OR name='downloads' OR name='segments' OR name='typed_url_sync_metadata');'
        Map queries found: 1. Processing queries...
Processing query 'Chrome Downloads'
         'Chrome Downloads' did not return any results. CSV will not be saved.
Processing map 'Google Chrome History Visits' with Id '1f37cc07-2b10-4c80-b845-84708e4f7429'
        Verifying database via 'SELECT count(*) FROM sqlite_master WHERE type='table' AND (name='urls' OR name='visits' OR name='downloads' OR name='segments' OR name='typed_url_sync_metadata');'
        Map queries found: 1. Processing queries...
Processing query 'History'
         'History' did not return any results. CSV will not be saved.
Processing map 'Google Chrome Keyword Searches' with Id 'd968ece4-d24a-4baf-8f07-78da07727712'
        Verifying database via 'SELECT count(*) FROM sqlite_master WHERE type='table' AND (name='urls' OR name='visits' OR name='downloads' OR name='segments' OR name='typed_url_sync_metadata');'
        Map queries found: 1. Processing queries...
Processing query 'Chrome Keyword Searches'
         'Chrome Keyword Searches' did not return any results. CSV will not be saved.

Processed 1 file in 0.6752 seconds
```

`did not return any results. CSV will not be saved` is what I would expect if the Map lined up with the schema of the DB. Thoughts?

**Comment by anelshaer** [comment]
I have shared that in my previous comment, i can share the db files with you just let me know the preferred method.

```
Processing 'C:\Users\user1\Desktop\testkape\C\Program Files\Bitdefender\Bitdefender Security\RansomwareRecover.db'...
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
         'Bitdefender RansomewareRecover DB Files' did not return any results. CSV will not be saved.
```

**Comment by rathbuna** [comment]
The output messages in your most recent message and my previous one are different. There's an issue with the Map. 

You can upload it here. If you're not able to, I'll give you an email.

**Comment by anelshaer** [comment]
here are the 3 DB files I'm creating a Map for, the rest of the DB files are also empty for me and did not yet create a Map for. 

what I found weird in these errors, is I have 3 Maps with `IdentifyValue`  1, 1, 2, and those what I'm getting as 0. altho the DB which have data is fully dumped as expected and the empty is not dumped so I'm getting the intended results. 
```
 For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '1'. Queries will not be processed!
        For map w/ description 'Bitdefender Sqlite Databases', got value '0' from IdentityQuery, but expected '2'. Queries will not be processed!
```
[Bitdefender Security.zip](https://github.com/EricZimmerman/SQLECmd/files/6536967/Bitdefender.Security.zip)

**Comment by rathbuna** [comment]
Thank you for providing those 3 DB's. I tested the 3 Maps in this Pull Request against those DBs and I got output indicating the Maps are valid and are working against the DB's, regardless of whether they have content or not. So we're squared away there. See output below:

```
Checking if 'E:\SQLECMD Maps\Antiphishing.db' is a SQLite file
'E:\SQLECMD Maps\Antiphishing.db' is a SQLite file!
Adding 'E:\SQLECMD Maps\Antiphishing.db' SHA-1 '93D2AAC104CF1A590864B6D387AB28E067B9F74C' to seen hashes collection
Processing 'E:\SQLECMD Maps\Antiphishing.db'...
Processing map 'Bitdefender Sqlite Antiphishing Database' with Id 'c3a9343a-cd65-484e-a8d2-b4e69b9924a1'
        Verifying database via 'SELECT count(*) FROM sqlite_master WHERE type='table' AND (name='aph_cache');'
        Map queries found: 1. Processing queries...
Processing query 'Bitdefender Antiphishing DB'
        Dumping 'Bitdefender Antiphishing DB' to 'E:\SQLECMD Maps\20210525101812820832_Bitdefender_Antiphishing_c3a9343a-cd65-484e-a8d2-b4e69b9924a1.csv'

Checking if 'E:\SQLECMD Maps\es.db' is a SQLite file
'E:\SQLECMD Maps\es.db' is a SQLite file!
Adding 'E:\SQLECMD Maps\es.db' SHA-1 '1002417882B3A0CCF466C2CC2DE33A98E56A6BB1' to seen hashes collection
Processing 'E:\SQLECMD Maps\es.db'...
Processing map 'Bitdefender Sqlite ES Database' with Id 'b5341baa-9fa9-4761-9a86-4fca69aca6e8'
        Verifying database via 'SELECT count(*) FROM sqlite_master WHERE type='table' AND (name='es_cache');'
        Map queries found: 1. Processing queries...
Processing query 'Bitdefender es DB'
        Dumping 'Bitdefender es DB' to 'E:\SQLECMD Maps\20210525101813115046_Bitdefender_es_b5341baa-9fa9-4761-9a86-4fca69aca6e8.csv'

Checking if 'E:\SQLECMD Maps\RansomwareRecover.db' is a SQLite file
'E:\SQLECMD Maps\RansomwareRecover.db' is a SQLite file!
Adding 'E:\SQLECMD Maps\RansomwareRecover.db' SHA-1 '42B035CFD58C4E7AEBBB1598723B530AA12DFB60' to seen hashes collection
Processing 'E:\SQLECMD Maps\RansomwareRecover.db'...
Processing map 'Bitdefender Sqlite RansomwareRecover Database' with Id 'e5b7cddd-edd9-4717-b51e-10a1b07bca61'
        Verifying database via 'SELECT count(*) FROM sqlite_master WHERE type='table' AND (name='files' OR name='packs');'
        Map queries found: 1. Processing queries...
Processing query 'Bitdefender RansomwareRecover DB Files'
         'Bitdefender RansomwareRecover DB Files' did not return any results. CSV will not be saved.
```

Before I merge this, I just want to clear up a bit of confusion I have. I'm seeing `For map w/ description 'Bitdefender Sqlite Databases` in your output above where you're getting the `0 found but expected 1` error. That appears to be a different Map than what you've provided in this PR because none of the 3 have that description. The descriptions of the Maps in this PR are: `Bitdefender Sqlite Antiphishing Database`, `Bitdefender Sqlite ES Database`, and `Bitdefender Sqlite RansomwareRecover Database`. 

So I'm not sure what that Map is supposed to be parsing, but the issue is with that Map, not any of the 3 you have here. Can you provide more clarity on that Map or is that a Map that's yet to come?

It looks like that Map and the Antiphishing DB Map were shooting errors in this https://github.com/EricZimmerman/SQLECmd/pull/31#issuecomment-846620266 that indicate there's an issue with the IdentityQuery and IdentityValue properties.

**Comment by anelshaer** [comment]
Sorry about the confusion, it seems I had versions of my Map in KAPE  Module,  the only change they had was a different description I guess.
they were generating those warnings. I should be creating other maps for the other DBs hopefully soon. 

Thanks again for taking the time to troubleshoot that with me.

**Comment by rathbuna** [comment]
Thank you as well for your patience.

**Pull request merged**